File: leetcode_c_plus_plus/495.cpp
// 495. 提莫攻击
// https://leetcode-cn.com/problems/teemo-attacking/
#include<map>
#include<set>
#include<vector>
#include<cmath>
#include<numeric>
using namespace std;
/**
@File : 495.cpp
@Time : 2021/11/10 22:19:52
@Author : YuMin Zhang
@State : Indepeedent | Depedent | Half-Depedent
@Thinking :
**/
class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int ans = 0;
        for(int i = 1; i < timeSeries.size(); i++) {
            if(timeSeries[i - 1 ] + duration - 1 >= timeSeries[i]) {
                ans += timeSeries[i] - timeSeries[i - 1];
            } else {
                ans += duration;
            }
        }    
        return ans + duration;    

    }
};

File: leetcode_python/495.py
# 495. 提莫攻击
# https://leetcode-cn.com/problems/teemo-attacking/
from typing import List
from collections import defaultdict
from collections import deque
from math import inf
import bisect
import heapq
'''
@File : 495.py
@Time : 2021/11/10 22:47:10
@Author : YuMin Zhang
@State : Indepeedent | Depedent | Half-Depedent
@Thinking :
'''
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        ans = 0
        experied = 0
        for i in timeSeries:
            if i >= experied:
                ans += duration
            else:
                ans += i + duration - experied
            experied = i + duration 
        return ans
            